File: JAVASCRIPT/linkedlist.js
function Node(val) {
  this.val = val;
  this.next = null;
}

function linkedlist() {
  var head_pointer = null;
  var total_len = 0;

  return {
     InsertAtHead(val2)
    {
    
      if (!val2) return;

      var node = new Node(val2);

      if (head_pointer) {
        node.next = head_pointer;
      }

      head_pointer = node;
      total_len++;
    },
    
    DeleteAtVal(val) {
      var current_pointer = head_pointer;

      if (head_pointer.val === val) {
        head_pointer = head_pointer.next;
        return;
      }

      while (current_pointer) {
        if (current_pointer.next) {
          var next = current_pointer.next;

          if (next.val === val) {
            current_pointer.next = next.next;
            total_len--;
            break;
          }
        }

        current_pointer = current_pointer.next;
      }
    },
    get len() {
      return total_len;
    },
    
    PrintingLinkedList() {
      var final = [];

      var current_pointer = head_pointer;
      while (current_pointer) {
        final.push(current_pointer.val);

        current_pointer = current_pointer.next;
      }

      return final;
      
    }
  }
}

File: PYTHON/linked_list.py
#python program for linked list traversal
class Node:

	def __init__(self, data):
		self.data = data 
		self.next = None 



class LinkedList:


	def __init__(self):
		self.head = None


	def printList(self):
		temp = self.head
		while (temp):
			print (temp.data)
			temp = temp.next



if __name__=='__main__':

	
	llist = LinkedList()

	llist.head = Node(1)
	second = Node(2)
	third = Node(3)

	llist.head.next = second; 
	second.next = third; 

	llist.printList()
